python/metatensor-operations/tests/is_contiguous.py

```diff
@@ -0,0 +1,69 @@
+import os
+
+import pytest
+
+import metatensor
+from metatensor import TensorBlock, TensorMap
+
+
+DATA_ROOT = os.path.join(os.path.dirname(__file__), "data")
+
+
+@pytest.fixture
+def tensor():
+    """Loads a TensorMap from file for use in tests"""
+    return metatensor.load(
+        os.path.join(DATA_ROOT, "qm7-power-spectrum.npz"),
+        use_numpy=True,
+    )
+
+
+@pytest.fixture
+def non_contiguous_tensor(tensor) -> TensorMap:
+    """
+    Make a TensorMap non-contiguous by reversing the order of the samples/properties
+    rows/columns in all values and gradient blocks.
+    """
+    keys = tensor.keys
+    new_blocks = []
+
+    for block in tensor.blocks():
+        new_block = _non_contiguous_block(block)
+        new_blocks.append(new_block)
+
+    return TensorMap(keys=keys, blocks=new_blocks)
+
+
+def _non_contiguous_block(block: TensorBlock) -> TensorBlock:
```

Comment:
Can we move this function above the `tensor` function where this one is called.

Reply:
and done!